src/metadatas.rs

```diff
@@ -225,16 +244,12 @@ fn compute_services_start_end_dates(gtfs: &gtfs_structures::Gtfs) -> HashMap<&st
 fn networks_start_end_dates(
     metadata: &Metadata,
     gtfs: &gtfs_structures::Gtfs,
-) -> HashMap<String, Option<HashMap<String, String>>> {
-    let format = |d: chrono::NaiveDate| d.format("%Y-%m-%d").to_string();
-    let result: HashMap<String, Option<HashMap<String, String>>> = if metadata.networks.len() == 1 {
+) -> HashMap<String, Option<Interval>> {
+    let result: HashMap<String, Option<Interval>> = if metadata.networks.len() == 1 {
         // if there is only one agency, get data from existing metadata
         let mut agency_start_end_dates = HashMap::default();
         let start_end = match (metadata.start_date.as_ref(), metadata.end_date.as_ref()) {
-            (Some(sd), Some(ed)) => Some(HashMap::from([
-                (String::from("start_date"), sd.to_owned()),
-                (String::from("end_date"), ed.to_owned()),
-            ])),
+            (Some(sd), Some(ed)) => Some(Interval(sd.parse().unwrap(), ed.parse().unwrap())),
```

Comment:
is it unsafe to unwrap here? It comes from our code, so it shouldn't fail. Is there a better way?

Reply:
It's not working because sd.parse() outputs a `Result`. If the unwrap is ok, I will let it that way :)
Thanks!

Reply:
you just need to add a `.ok()` after the parse (sorry, should have compiled it :wink: )